ruler_alertmanager_config adaption and documentation.

ToonTijtgat2 opened the pull request:
**What this PR does / why we need it**:
add documentation of ruler_alertmanager_config and make it possible to only fill in what is diffent from ruler config

The documentation is handy to know that this feature is existing.
The changes on the applyAlertmanagerDefaults function are handy to only put in the override what you want to override instead of having to fill in the default values over and over. (making the runtime config very dirty and long.)

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:
Thanks for reviewing this pull request.
We started using the ruler_alertmanager_config and saw that there was no documentation and try to find some inprovements.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/loki?pullRequest=11899) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/loki?pullRequest=11899) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/loki?pullRequest=11899) it.</sub>

JStickler commented:
@ToonTijtgat2 the file `/docs/sources/configure/_index.md` is automatically generated from the file `docs/sources/configure/index.template`.  Could you please make your docs changes in the template instead of the _index.md file.

ToonTijtgat2 commented:
> @ToonTijtgat2 the file `/docs/sources/configure/_index.md` is automatically generated from the file `docs/sources/configure/index.template`. Could you please make your docs changes in the template instead of the _index.md file.

Dear

Thanks for checking, I removed the changes on the _index.md file and added the doc description in the config of limits.go

Toon Tijtgat
